solr/core/src/test/org/apache/solr/cluster/VersionTrackerImpl.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.cluster;
+
+import org.apache.solr.common.util.TimeSource;
+import org.apache.solr.util.TimeOut;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class VersionTrackerImpl implements VersionTracker {
+    private int version = 0;
+
+    @Override
+    public synchronized void increment() {
+        version++;
+        this.notifyAll();
+    }
+
+    @Override
+    public int waitForVersionChange(int currentVersion, int timeoutSec) throws InterruptedException, TimeoutException {
+        TimeOut timeout = new TimeOut(timeoutSec, TimeUnit.SECONDS, TimeSource.NANO_TIME);
+        int newVersion = currentVersion;
+        while (! timeout.hasTimedOut()) {
```

Comment:
formatting: remove space after `!`

Reply:
I also realize I would have put the while loop inside the synchronized block, and now I wonder which way is better... Re-acquiring it over and over again or not holding it while seeing if a timeout occurred.

Reply:
It's hard to do profiling on this since it generally only loops once on my machine, but I'll switch it to the loop inside of the block. It's part of the condition that we are implicitly checking with the wait, so it makes sense this way too.